main.cpp

```diff
@@ -24,408 +22,153 @@
  
  7) make sure it compiles & runs without errors.
  
- 8) use your knowledge of casting to remove any conversion warnings. 
+ 8) use your knowledge of casting to remove any conversion warnings.
  
  Commit your changes by clicking on the Source Control panel on the left, entering a message, and click [Commit and push].
  
 Make a pull request after you make your first commit and pin the pull request link to our DM thread.
-
 send me a DM to check your pull request
-
  Wait for my code review.
  */
 
-
-
-
 #include <iostream>
 
-struct DoubleType;
-struct IntType;
-
 struct FloatType
 {
-    FloatType(float val) : value(new float(val)) {}
-    ~FloatType() { delete value; }
+    FloatType(float val = 0.0f) : m_value(new float(val)) {}
+    ~FloatType()
+	{
+		delete m_value;
+		m_value = nullptr;
+	}
 
-    float* value;
+	operator float() { return *m_value; }
 
-    // primitives
     FloatType& add(float v);
     FloatType& subtract(float v);
     FloatType& multiply(float v);
     FloatType& divide(float v);
-    // FloatType args
-    FloatType& add(const FloatType& ft);
-    FloatType& subtract(const FloatType& ft);
-    FloatType& multiply(const FloatType& ft);
-    FloatType& divide(const FloatType& ft);
-    // DoubleType args
-    FloatType& add(const DoubleType& dt);
-    FloatType& subtract(const DoubleType& dt);
-    FloatType& multiply(const DoubleType& dt);
-    FloatType& divide(const DoubleType& dt);
-    // IntType args
-    FloatType& add(const IntType& it);
-    FloatType& subtract(const IntType& it);
-    FloatType& multiply(const IntType& it);
-    FloatType& divide(const IntType& it);
+	
+private:
+	float* m_value { nullptr };
 };
 
 struct DoubleType
 {
-    DoubleType() {}
-    DoubleType(double val) : value(new double(val)) {}
-    ~DoubleType() { delete value; }
+    DoubleType(double val = 0.0) : m_value(new double(val)) {}
+    ~DoubleType()
+	{
+		delete m_value;
+		m_value = nullptr;
+	}
 
-    double* value;
+	operator double() { return *m_value; }
 
-    // primitives
     DoubleType& add(double v);
     DoubleType& subtract(double v);
     DoubleType& multiply(double v);
     DoubleType& divide(double v);
-    // FloatType args
-    DoubleType& add(const FloatType& ft);
-    DoubleType& subtract(const FloatType& ft);
-    DoubleType& multiply(const FloatType& ft);
-    DoubleType& divide(const FloatType& ft);
-    // DoubleType args
-    DoubleType& add(const DoubleType& dt);
-    DoubleType& subtract(const DoubleType& dt);
-    DoubleType& multiply(const DoubleType& dt);
-    DoubleType& divide(const DoubleType& dt);
-    // IntType args
-    DoubleType& add(const IntType& it);
-    DoubleType& subtract(const IntType& it);
-    DoubleType& multiply(const IntType& it);
-    DoubleType& divide(const IntType& it);
+	
+private:
+	double* m_value { nullptr };
```

Comment:
- [ ] FIXME: don't use `m_` in variable names.  Why did you change this?  You're just supposed to make it private, not rename it.